From 6ef45bbe919a3cdf60fb95b6847c08acec96dd1a Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Mon, 22 May 2023 16:57:32 +0200
Subject: [PATCH 1/2] sound: soc: codecs: spdif_transmitter: less restrictions,
 fix dapm

Don't force DAPM bias to always on (messes with other components)
Allow any PCM rate, not just a standard set.
---
 sound/soc/codecs/spdif_transmitter.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/sound/soc/codecs/spdif_transmitter.c b/sound/soc/codecs/spdif_transmitter.c
index 2c8cebfc6603..81ce317ba904 100644
--- a/sound/soc/codecs/spdif_transmitter.c
+++ b/sound/soc/codecs/spdif_transmitter.c
@@ -21,7 +21,7 @@
 
 #define DRV_NAME "spdif-dit"
 
-#define STUB_RATES	SNDRV_PCM_RATE_8000_192000
+#define STUB_RATES	SNDRV_PCM_RATE_CONTINUOUS
 #define STUB_FORMATS	(SNDRV_PCM_FMTBIT_S16_LE | \
 			SNDRV_PCM_FMTBIT_S20_3LE | \
 			SNDRV_PCM_FMTBIT_S24_LE  | \
@@ -40,7 +40,6 @@ static struct snd_soc_component_driver soc_codec_spdif_dit = {
 	.num_dapm_widgets	= ARRAY_SIZE(dit_widgets),
 	.dapm_routes		= dit_routes,
 	.num_dapm_routes	= ARRAY_SIZE(dit_routes),
-	.idle_bias_on		= 1,
 	.use_pmdown_time	= 1,
 	.endianness		= 1,
 	.non_legacy_dai_naming	= 1,
-- 
2.17.1

